force-app/main/default/classes/IntegrationConfig.cls

```diff
@@ -0,0 +1,28 @@
+public class IntegrationConfig {
+
+    public String host;
+    public String method;
+    public String namedCredentials;
+    public String path;
+    public Integer timeout;
+
+    public static IntegrationConfig forName(String devName) {
+        Integration_Config__mdt config = fetchIntegrationConfigs(devName);
+        
+        return new IntegrationConfig(config);
+    }
+
+    public IntegrationConfig(Integration_Config__mdt config){
+        this.host = config.Host__c;
+        this.method = config.Method__c;
+        this.namedCredentials = config.Named_Credentials__c;
+        this.path = config.Path__c;
+        this.timeout = config.Timeout__c?.intValue();
```

Comment:
First time learning about the null safe operator. Amazing! Prevents laziness induced errors because I didn't feel writing an if null check.